lib/jekyll/converters/markdown/kramdown_parser.rb

```diff
@@ -47,6 +49,22 @@ def convert(content)
         end
 
         private
+
+        def load_dependencies
+          return if Kramdown::VERSION.to_i < 2
+          require "kramdown-parser-gfm" if @config["input"] == "GFM"
```

Comment:
Since `kramdown-parser-gfm` is not bundled by default, we should gracefully fail here. Gemfile would have the gem listed only for new ones bootstrapped via `jekyll new`
```suggestion
          if @config["input"] == "GFM"
            Jekyll::External.require_with_graceful_fail("kramdown-parser-gfm")
          end
```